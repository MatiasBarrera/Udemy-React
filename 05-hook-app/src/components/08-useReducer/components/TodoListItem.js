import React from 'react'
import PropTypes from 'prop-types'

const TodoListItem = ({ todo, index, handleToggle, handleDelete }) => {
  return (
    <li key={todo.id} className="list-group-item">
      <p
        className={`${todo.done && 'complete'}`}
        onClick={() => handleToggle(todo.id)}>
        {index + 1}. {todo.desc}
      </p>
      <button className="btn btn-danger" onClick={() => handleDelete(todo.id)}>
        Borrar
      </button>
    </li>
  )
}

TodoListItem.propTypes = {
  todo: PropTypes.object,
  index: PropTypes.number,
  handleToggle: PropTypes.func.isRequired,
  handleDelete: PropTypes.func.isRequired
}

export default TodoListItem
