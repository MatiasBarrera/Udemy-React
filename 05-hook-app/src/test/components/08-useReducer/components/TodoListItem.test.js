import React from 'react'
import { shallow } from 'enzyme'
import TodoListItem from '../../../../components/08-useReducer/components/TodoListItem'
import { demoTodos } from '../../../fixtures/demoTodos'

describe('Pruebas en TodoListItem', () => {
  const handleDelete = jest.fn()
  const handleToggle = jest.fn()

  const wrapper = shallow(
    <TodoListItem
      todos={demoTodos[0]}
      handleToggle={handleToggle}
      handleDelete={handleDelete}
    />
  )

  test('Debe de mostrar el componente correctamente', () => {
    expect(wrapper).toMatchSnapshot()
  })

  test('Debe de llamar a la funcion handleDelete', () => {
    wrapper.find('button').simulate('click')
    expect(handleDelete).toHaveBeenCalledWith(demoTodos[0].id)
  })

  test('Debe de llamar a la funcion handleToggle', () => {
    wrapper.find('p').simulate('click')
    expect(handleToggle).toHaveBeenCalledWith(demoTodos[0].id)
  })

  test('Debe de mostrar el texto correctamente', () => {
    const p = wrapper.find('p').text().trim()
    expect(p).toBe(`1 ${demoTodos[0].desc}`)
  })

  test('Debe de tener la clase complete si el TODO.done = true', () => {
    const todo = demoTodos[0]
    todo.done = true
    const wrapper = shallow(
      <TodoListItem
        todos={todo}
        handleToggle={handleToggle}
        handleDelete={handleDelete}
      />
    )

    expect(wrapper.find('p').hasClass('complete')).toBeTruthy()
  })
})
