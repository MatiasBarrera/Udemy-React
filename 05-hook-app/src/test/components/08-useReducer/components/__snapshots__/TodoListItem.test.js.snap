// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en TodoListItem Debe de mostrar el componente correctamente 1`] = `
<Fragment>
  <li
    className="list-group-item"
    key="1"
  >
    <p
      className="false"
      id={1}
      onClick={[MockFunction]}
    >
      1
       
      Aprender react
    </p>
    <button
      className="btn btn-danger"
      id={1}
      onClick={[MockFunction]}
      type="button"
    >
      -
    </button>
  </li>
</Fragment>
`;
